### DISCUSSION TITLE
Jotai v2: why not keep/add initialValues API

### DISCUSSION THREAD

**Discussion by adaptive-shield-matrix** [pmndrs/jotai#1788] [General] [answered]
Currently I have to use a workaround, to use the same Provider API.
Would it not be beneficial to simplify the transition to v2 for all devs by allowing to use old API methods?
Am I missing something? 

My utility function
```ts
export function createJotaiStore(pairs: [PrimitiveAtom<any>, any][]): JotaiStore {
  const s = createStore()
  for (let pair of pairs) {
    const [atom, v] = pair
    s.set(atom, v)
  }
  return s
}
```

example code
```ts
  const store = createJotaiStore([
    // general
    [notistackAtom, { enqueue: enqueueSnackbar, close: closeSnackbar }],
    [routerAtom, { push: router.push, pathname: router.pathname, query: router.query }],
    [sessionAtom, session],
    [isDevAtom, isDev],
    [isAdminAtom, isAdmin],
    [isGroupOrganizerModeAtom, isGroupOrganizerMode],
    // saving
    [savedTripAtom, trip],
    [savedUiSettingsAtom, settings ?? null],
    // trip
    [tidAtom, tid],
    [tripNameAtom, trip.name],
  ])


  return (
    <Provider store={store}>
      <TripStepPage />
    </Provider>
  )
```

**Comment by dai-shi** [comment, marked as answer]
In v2, the recommendation is to use `useHydrateAtoms` which is more flexible. Maybe there's room for improvement in the migration guide.

As for API design, we try to avoid providing multiple ways (this applies not only to jotai but also to other state libs in pmndrs). So, creating a util on your end (or in a third-party library) is very expected.
